src/status_im/ui/screens/wallet/send/events.cljs

```diff
@@ -160,21 +161,34 @@
            ::show-transaction-error error_message}
           {:db (update-in db [:wallet :transactions-unsigned] dissoc id)})))))
 
+(defn prepare-unconfirmed-transaction [db now hash id]
+  (-> (get-in db [:wallet :transactions-unsigned id])
+      (assoc :confirmations "0"
+             :timestamp (str now)
+             :type :outbound
+             :hash hash)
+      (update :gas-price str)
+      (update :value str)
+      (update :gas str)
+      (dissoc :message-id :id)))
+
 (handlers/register-handler-fx
   ::transaction-completed
-  (fn [{db :db} [_ {:keys [id response]} modal?]]
+  (fn [{db :db now :now} [_ {:keys [id response]} modal?]]
     (let [{:keys [hash error]} response
           db' (assoc-in db [:wallet :send-transaction :in-progress?] false)]
       (if (and error (string? error) (not (string/blank? error))) ;; ignore error here, error will be handled in :transaction-failed
         {:db db'}
-        (merge
-          {:db (-> db'
-                   (update-in [:wallet :transactions-unsigned] dissoc id)
-                   (update-in [:wallet :send-transaction] merge clear-send-properties))}
-          (if modal?
-            {:dispatch-n [[:navigate-back]
-                          [:navigate-to-modal :wallet-transaction-sent-modal]]}
-            {:dispatch [:navigate-to :wallet-transaction-sent]}))))))
+        (let [hash (:hash response)]
```

Comment:
Did we already capture `hash` in `let` on line 178 ?

Reply:
thanks good catch !